src/sync-commands/overwrite-all-remote-with-local.js

```diff
@@ -35,11 +41,11 @@ class OverwriteAllRemoteWithLocal {
 
     async handleAllFiles() {
         for (const h of this.diff.filesThatDiffer) {
-            await this.updateItem(h);
+            this.asyncPool.add(() => this.updateItem(h));
         }
 
         for (const h of this.diff.extraItems.local.files) {
-            await this.createItem(h);
+            this.asyncPool.add(() => this.createItem(h));
         }
     }
```

Comment:
This used to be a sync call with the awaits but now is using a pool

Should there be something to wait for the asyncPool to finish at the end of this function?

Reply:
It is correct. We put tasks into the `pool` and these tasks will be executed later according to the limits: 100 per minute, 4 tasks in parallel.

Reply:
I think the issue here is that there is no longer a known point when the tasks are done with this change. Before, calling handleAllFiles() would have finished the update of items before exiting?

Now with the asyncPool, is there a possibility of race conditions, especially using `watch` mode?

Reply:
I'll fix it. `this.asyncPool.add`  will return promise. `handleAllFiles` will collect them, push them into array, and return `Promise.all([...])`